### ISSUE TITLE
Broadcast a signal using the gRPC API

### ISSUE THREAD

**Issue by korthout** [camunda/zeebe#11196]
Once top-level signal start events can be subscribed to signals, we need to make it possible to broadcast signals using the gRPC API.

We'll do this using a new `Signal` record. When the gateway receives a signal broadcast, it can `Broadcast` it to the deployment partition (i.e. `partitionId: 1`). 

These tasks belong together, but it's probably best to split them into multiple pull requests.

- [x] https://github.com/camunda/zeebe/issues/11295 Introduce new `Signal` records, with intents: `Broadcast`, `Broadcasted`
	- The signal record value should consist of:
		- `signalName`
		- optionally `variables`
		- Like any other records, these should be exported. Please follow our [developer handbook entry on adding new records](https://github.com/camunda/zeebe/blob/main/docs/developer_handbook.md#how-to-create-a-new-record)
- [x] #11777 Introduce new `Signal:Broadcast` command processor
	- Writes a `Signal:Broadcasted` event indicating that the command was processed successfully, otherwise the command is rejected. No event applier is needed for it.
	- For each `SignalSubscription` matching the `signalName`, uses `DbSignalSubscriptionState.visitBySignalName`:
		- If related catch event is a top-level start event:
			- Start an instance of the latest version of the process defined by `bpmnProcessId`, at start event `catchEventId` 
- [x] https://github.com/camunda/zeebe/issues/11878 Introduce new broadcast signal RPC in gateway-protocol
	- The RPC takes a message containing:
		- `signalName`
		- optionally `variables`
	- Implementation should write a `Signal:Broadcast` command to the deployment partition leader
		- Can start with deployment partition
		- Round-robin signal broadcasts from the gateway (optimization, stretch goal)
- [x] https://github.com/camunda/zeebe/issues/11912 Introduce a java-client implementation of new RPC
	- Should be straightforward
- [x] https://github.com/camunda/zeebe/issues/11913 Introduce a go-client implementation of new RPC
	- Should be straightforward

Blocked by
- #11195

**Comment by korthout** [comment]
Hi @skayliu 👋 Thanks again for your previous contributions on this topic 🙇 

Would you be willing to contribute to the other parts of this feature? I believe it is the last part that is needed to get #11194 into the next alpha release `8.2.0-alpha5`.

**Comment by lzgabel** [comment]
Hi @korthout. Since @skayliu is busy with work recently, I will take over the follow-up work. I'll be working on it later this week or next week. ❤️

**Comment by korthout** [comment]
Thanks for letting me know and for taking over. Looking forward to it. 🚀

**Comment by skayliu** [comment]
Hi, @korthout, Recently I'm busy with work, Sorry for the delay, I'll co-work with @lzgabel for the last part.

**Comment by korthout** [comment]
@skayliu Don't worry if other work comes in between. Thanks for all the efforts you put into this already ❤️ We really appreciate it.

**Comment by lzgabel** [comment]
Hi @korthout. I've completed all the remaining tasks, and I'll submit a PR for review next week (Blocked by #11778). (Blocked by #11778). Hope you can take the time to review next week. :heart:

**Comment by lzgabel** [comment]
👋 Hi @korthout. All the tasks have been completed, we can close this issue, thanks for the review. :heart:

**Comment by korthout** [comment]
Great news! Thanks for all the contributions! 🏅

**Comment by lzgabel** [comment]
❓ @korthout. Sorry to bother you again :bow:.  A small question, the tag of `8.2.0-alpha5` was created yesterday, will the client part that has just been merged be included in this version?

**Comment by korthout** [comment]
@lzgabel No, sorry. The `8.2.0-alpha5` tag was made from a branch that we froze a week ago. `8.2.0-alpha5` does not have the broadcast signal RPC in the gateway yet.
